wordpress-mobile/WordPress-Android
# Conflict Resolution: Pages

The PR adds support for resolving page conflict resolutions
- Version conflicts 
- Autosave conflicts
The new PostConflictResolutionOverlay is shown in place of the older Android dialog. Note: Other dialogs remain as is.

Specs pcdRpT-5ID-p2#comment-9638.

-----
## To Test:
- Install the Jetpack and WordPress apps from this PR

### Test Page version forced write on top of web (existing logic)
- Login with a site that has pages
- Navigate to Me > Debug Settings
- Disable the `sync_publishing` flag and restart the app
- Navigate to back to My Site > Pages
- Tap the publish tab and then tap a page to open the edit view
- PAUSE - GO TO YOUR LAPTOP
- Open a browser and navigate to the same WordPress site
- Open the same page and make an edit (remember what you changed)
- Tap the "update" button to save it and back out of the editor
- GO BACK TO YOUR DEVICE
- Make a change to the page you have in the edit view
- Tap "Update" to save the page to the server
- ✅ Verify the page did update (snackbar)
- GO BACK TO THE LAPTOP
- Tap view on the same page you just edited via the app
- ✅ Verify the changes you made on the device have overwritten the changes you made on your laptop
- Repeat this test for Jetpack and Wordpress

### Test Page version did not write on top of web (new logic)
- Navigate to Me > Debug Settings
- Enabled the `sync_publishing` flag and restart the app 
- Navigate to back to My Site > Pages
- Tap the publish tab and then tap a page to open the edit view
- PAUSE - GO TO YOUR LAPTOP
- Open a browser and navigate to the same WordPress site
- Open the same page and make an edit (remember what you changed)
- Tap the "update" button to save it and back out of the editor
- GO BACK TO YOUR DEVICE
- Make a change to the page you have in the edit view
- Tap "Update" to save the page to the server
- ✅ Verify the page did not update and the post list shows "Version Conflict" within the post row
- GO BACK TO THE LAPTOP
- Tap view on the same page you just edited via the app
- ✅ Verify the changes you made on the device did not overwrite the changes you made on your laptop
- Repeat this test for Jetpack and Wordpress

### Test Conflict Resolution dialog is shown
- Continue from the last step of test "Page version did not write on top of web (new logic)"
- On the Pages list, tap on the page that shows "Version Conflict"
- ✅ Verify the conflict resolution dialog is shown
- Tap local version
- ✅ Verify the local version has been applied
- Repeat the scenario in  test "Page version did not write on top of web (new logic)" to get into a Version Conflict again. Then repeat this test and tap web version in the conflict resolution dialog.


-----

## Regression Notes

1. Potential unintended areas of impact
The page is not updated properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Updated `PagesViewModelTest`, `PageListViewModelTest`, and `CreatePageListItemActionsUseCaseTest`

3. What automated tests I added (or what prevented me from doing so)
Relied on existing instrumentation tests

-----

## PR Submission Checklist:

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
